cli/pcluster/config/param_types.py

```diff
@@ -0,0 +1,1202 @@
+            cfn_params.update({cfn_converter: str(cfn_value)})
+
+        return cfn_params
+
+    def get_default_value(self):
+        """Get default value from the Param map if there, None otherwise."""
+        return self.map.get("default", None)
+
+    def get_cfn_value(self):
+        """
+        Convert parameter value into CFN value.
+
+        Used when the parameter must go into a comma separated CFN parameter.
+        """
+        return str(self.value if self.value is not None else self.map.get("default", "NONE"))
+
+
+class FloatParam(Param):
+    """Class to manage float configuration parameters."""
+
+    def _init_from_file(self, config_parser):
+        """
+        Initialize param_value from config_parser.
+
+        :param config_parser: the configparser object from which get the parameter
+        :raise NoOptionError, NoSectionError if unable to get the param from config_parser
+        """
+        section_name = _get_file_section_name(self.section_key, self.section_label)
+        try:
+            self.value = config_parser.getfloat(section_name, self.key)
+            self._check_allowed_values()
+        except ValueError:
+            error("Configuration parameter '{0}' must be a Float".format(self.key))
+
+    def get_value_from_string(self, string_value):
+        """Return internal representation starting from CFN/user-input value."""
+        param_value = self.get_default_value()
+
+        try:
+            if string_value is not None and isinstance(string_value, str):
+                string_value = string_value.strip()
+                if string_value != "NONE":
+                    param_value = float(string_value)
+        except ValueError:
+            pass
+
+        return param_value
+
+
+class BoolParam(Param):
+    """Class to manage boolean configuration parameters."""
+
+    def _init_from_file(self, config_parser):
+        """
+        Initialize param_value from config_parser.
+
+        :param config_parser: the configparser object from which get the parameter
+        :raise NoOptionError, NoSectionError if unable to get the param from config_parser
+        """
+        section_name = _get_file_section_name(self.section_key, self.section_label)
+        try:
+            self.value = config_parser.getboolean(section_name, self.key)
+            self._check_allowed_values()
+        except ValueError:
+            error("Configuration parameter '{0}' must be a Boolean".format(self.key))
+
+    def get_value_from_string(self, string_value):
+        """Return internal representation starting from CFN/user-input value."""
+        param_value = self.get_default_value()
+
+        if string_value is not None and isinstance(string_value, str):
+            string_value = string_value.strip()
+            if string_value != "NONE":
+                param_value = string_value == "true"
+
+        return param_value
+
+    def to_file(self, config_parser):
```

Comment:
why can't we use the one from the base class?

Reply:
Right, done